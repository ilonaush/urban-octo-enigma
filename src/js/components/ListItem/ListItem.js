import React, {Component} from 'react';
import PropTypes from 'prop-types';
import "./ListItem.styl";
import MaskedInput from 'react-text-mask'

class ListItem extends Component {
     constructor(props) {
         super(props);
         this.state = {
             editing: ''
         };
         this.editTime = this.editTime.bind(this);
         // this.handleChange = this.handleChange.bind(this);
         this.saveTime = this.saveTime.bind(this);
     }

    editTime(e) {
        this.setState({
            editing: e.currentTarget.id
        })
    }

    saveTime({target}) {
        this.setState({
            editing: ''
        });
        const worker = {
            ...this.props.worker,
            [target.name]: target.value
        };
        this.props.editWorkerTime(worker);
    }

    render() {
     const {editing} = this.state;
     const {worker: {fullname, position, arrival = '', leaving = ''} = {}} = this.props;
        return (
            <tr>
                <td>
                    {fullname}
                </td>
                <td>
                    {position}
                </td>
                <td id='arrival' onClick={this.editTime} onBlur={this.saveTime} className='worker-time'>
                    {editing === 'arrival' ? <MaskedInput
                        mask={[' ', /\d/,  /\d/, ':',  /\d/,  /\d/]}
                        className='time-input'
                        name='arrival'
                        guide={true}
                        // onChange={this.handleChange}
                        placeholder='––:––'
                    />
                    : arrival ? arrival : 'Click to select'}
                </td>
                <td id='leaving' onClick={this.editTime} onBlur={this.saveTime} className='worker-time'>
                    {editing === 'leaving' ? <MaskedInput
                        mask={[' ', /\d/,  /\d/, ':',  /\d/,  /\d/]}
                        className='time-input'
                        name='leaving'
                        guide={true}
                        // onChange={this.handleChange}
                        placeholder='––:––'
                    />
                    : leaving ? leaving : 'Click to select'}
                </td>
            </tr>
        );
    }
}

ListItem.propTypes = {};

export default ListItem;
