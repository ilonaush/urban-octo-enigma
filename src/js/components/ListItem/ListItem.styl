@import "~js/vars"

button.btn-action
  width 130px
  margin 0
  padding 0

.table-row-item
  a
    color $blue
    font-weight bold
    text-decoration none
    transition 0.3s color
    &:hover
      color $violet

