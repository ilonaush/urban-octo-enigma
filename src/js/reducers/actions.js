export default {
    employWorker: function (worker) {
        return {
            type: 'EMPLOY_WORKER',
            worker: worker
        }
    },
    editWorkTime: function (worker) {
        return {
            type: 'EDIT_WORK_TIME',
            worker: worker,
        }
    },
    fireWorker: function (workerId) {
        return {
            type: 'FIRE_WORKER',
            id: workerId
        }
    }
};
