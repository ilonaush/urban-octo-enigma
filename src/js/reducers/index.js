import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer   from './reducers';
import  { saveStore, loadStore } from '../services/LocalStorageService';

const predefinedStore = loadStore() || {};

const store = createStore(
    reducer,
    predefinedStore

);

store.subscribe(() => {
    saveStore(store)
});

export {
    store, predefinedStore
};
