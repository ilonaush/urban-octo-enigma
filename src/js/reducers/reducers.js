import {ACTIONS_TYPES} from "../constants";

const initialState = {
  loading: false,
};

let reducer = (state = initialState, action) => {
  switch (action.type) {
    case ACTIONS_TYPES.GET_CATS:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.ADD_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.FEED_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.HUG_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.WASH_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.ISSUE_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.SET_LOADING: {
      return {
        ...state,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      }
    }
    default:
      return state;
  }
};

export default reducer;
