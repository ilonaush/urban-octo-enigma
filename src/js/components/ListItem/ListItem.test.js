import React from 'react';
import ReactDOM from 'react-dom';
import {ListItem} from './ListItem';
import { mount } from 'enzyme';
import {Provider} from "react-redux";
import {store} from "../../reducers";

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Provider  store={store}><ListItem/></Provider>, div);
});

it('does not render button', () => {
    const cat =  {
        feedingTime: '2019-03-14T10:59:53+02:00'
    };
    const ListItemComponent = mount(<Provider store={store}><ListItem cat={cat}/></Provider>);
    expect(ListItemComponent.find('#feeding-btn').exists()).toBeFalsy();
});

it('renders button', () => {
    const ListItemComponentButton = mount(<Provider store={store}><ListItem/></Provider>);
    console.log(ListItemComponentButton.props());
    expect(ListItemComponentButton.find('#feeding-btn').exists()).toBeTruthy();
});
