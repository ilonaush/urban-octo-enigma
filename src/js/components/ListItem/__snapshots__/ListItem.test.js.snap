// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders without crashing 1`] = `
<div
  className="table-row"
>
  <div
    className="table-row-item"
  >
    <Link
      replace={false}
      to="/cat/undefined"
    />
  </div>
  <div
    className="table-row-item"
  >
    -
  </div>
  <div
    className="table-row-item"
  >
    -
  </div>
  <Connect(CatManipulationPanel)
    cat={Object {}}
  />
</div>
`;
